rma_sale/static/src/js/rma_portal_form.js

```diff
@@ -4,42 +4,122 @@
 odoo.define("rma_sale.animation", function(require) {
     "use strict";
 
-    var sAnimation = require("website.content.snippets.animation");
+    const publicWidget = require("web.public.widget");
 
-    // In the customer portal when a RMA operation is selected show the comments
-    // selector so the user doesn't miss the chance to add his comments
-    sAnimation.registry.rma_operation_portal = sAnimation.Class.extend({
-        selector: ".rma-operation",
+    /**
+     * Adds some machinery to the customer portal RMA form:
+     *
+     * - Avoid submitting the form if no qty or operation is reported.
+     * - Show automatically the observations field when the operation is selected
+     *   and hide it back with no operation selected.
+     */
+    publicWidget.registry.PortalRmaSale = publicWidget.Widget.extend({
+        selector: "#form-request-rma",
+        events: {
+            "change .rma-operation": "_onChangeOperationId",
+            "change #delivery-rma-qty input": "_onChangeQty",
+        },
+
+        /**
+         * @override
+         */
         start: function() {
-            this.id = this.el.name.replace("-operation_id", "");
-            this.$comment = $("#comment-" + this.id);
-            this.$comment_input = $("[name='" + this.id + "-description']");
+            const ids = $("[name*='-operation_id']")
+                .map(function() {
+                    return this.name.replace("-operation_id", "");
+                })
+                .get();
+            this.$submit = $("#form-request-rma button[type='submit']");
+            this.rows_ids = ids;
+            // We'll build an object that will ease the form check. It could be further
+            // extended with additional checks.
+            this.rows = {};
             var _this = this;
-            this.$el.on("change", function() {
-                _this._onChangeOperationId();
+            _.each(ids, function(id) {
+                _this.rows[id] = {};
+                _this.rows[id].$comment = $("#comment-" + id);
+                _this.rows[id].$comment_input = $("[name='" + id + "-description']");
+                _this.rows[id].$operation = $("[name='" + id + "-operation_id']");
+                _this.rows[id].$qty = $("[name='" + id + "-quantity']");
             });
```

Comment:
It might be interesting to use es6... You may notice that "_this" disappears since the function will use the same context as the one invoking it. Also, for better readability "template strings" is used.
```suggestion
            _.each(ids, (id) => {
                this.rows[id] = {
                    $comment: this.$(`#comment-${id}`),
                    $comment_input: this.$(`[name='${id}-description']`),
                    $operation: this.$(`[name='${id}-operation_id']`),
                    $qty: this.$(`[name='${id}-quantity']`)
                };
            });
```